pkg/trace/config/apply.go

```diff
@@ -253,7 +253,9 @@ func (c *AgentConfig) applyDatadogConfig() error {
 	c.TraceWriterConfig = readTraceWriterConfigYaml()
 
 	// allow an extra 10% of the maximum number of connections for parallel writes to the Datadog API
-	c.TraceWriterConfig.SenderConfig.MaxConnections = int(math.Max(1, float64(c.ConnectionLimit)/10))
+	conns5percent := int(math.Max(1, float64(c.ConnectionLimit)/20))
+	c.TraceWriterConfig.SenderConfig.MaxConnections = conns5percent
+	c.StatsWriterConfig.SenderConfig.MaxConnections = conns5percent
```

Comment:
What about `c.ServiceWriterConfig`?

Reply:
I think that's low volume and we can safely leave it as is.